math/fibonacci_sum.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @brief An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+ * @details An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$ where \f$\mathrm{F}(i)\f$
+ * denotes the i-th Fibonacci Number . Note that F(0) = 0 and F(1) = 1.
+ * The value of the sum is calculated using matrix exponentiation.
+ * @source https://stackoverflow.com/questions/4357223/finding-the-sum-of-fibonacci-numbers
+ * @author [Sarthak Sahu](https://github.com/SarthakSahu1009)
+ */
+
+#include <iostream> /// for std::cin and std::cout
+#include <cassert>  /// for assert
+#include <vector>   /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+    /**
+     * @namespace fibonacci_sum
+     * @brief Functions for the sum of the Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+     */
+    namespace fibonacci_sum {
+
+        /**
+         * Function to multiply two matrices
+         * @param T matrix 1
+         * @param A martix 2
+         * @returns resultant matrix
+         */
+        std::vector<std::vector<int> > multiply(std::vector<std::vector<int> > T, std::vector<std::vector<int> > A) {
+
+            std::vector<std::vector<int> > result(2,std::vector<int>(2));
```

Comment:
```suggestion
            std::vector<std::vector<int> > result(2,std::vector<int>(2, 0));
```
this initializes the vector elements to value '0'